apps/token/src/app-loader.tsx

```diff
@@ -0,0 +1,157 @@
+import * as Sentry from '@sentry/react';
+import { Splash } from '@vegaprotocol/ui-toolkit';
+import { useVegaWallet } from '@vegaprotocol/wallet';
+import { useWeb3React } from '@web3-react/core';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { SplashError } from './components/splash-error';
+import { SplashLoader } from './components/splash-loader';
+import { Flags } from './config';
+import {
+  AppStateActionType,
+  useAppState,
+} from './contexts/app-state/app-state-context';
+import { useContracts } from './contexts/contracts/contracts-context';
+import { useRefreshAssociatedBalances } from './hooks/use-refresh-associated-balances';
+import { getDataNodeUrl } from './lib/get-data-node-url';
+
+export const AppLoader = ({ children }: { children: React.ReactElement }) => {
+  const { t } = useTranslation();
+  const { account } = useWeb3React();
+  const { keypair } = useVegaWallet();
+  const { appDispatch } = useAppState();
+  const { token, staking, vesting } = useContracts();
+  const setAssociatedBalances = useRefreshAssociatedBalances();
+  const [balancesLoaded, setBalancesLoaded] = React.useState(false);
+  // TODO: TFE import detect when vega keys attempt complete
```

Comment:
?

Reply:
- Moved eager connection hook for vega wallet to libs/wallet
- Added some logic to return a connecting state from the hook which we can use to prevent rendering until the initial keys have been fetched restoring prior functionality